cms/server/contest/handlers/base.py

```diff
@@ -221,13 +221,14 @@ def get_user_locale(self):
             lang_codes = filter_language_codes(
                 lang_codes, self.contest.allowed_localizations)
 
-        # TODO We fallback on "en" if no language matches: we could
+        # TODO We fallback on basic_lang if no language matches: we could
         # return 406 Not Acceptable instead.
         # Select the one the user likes most.
+        basic_lang = 'en' if 'en' in lang_codes else lang_codes[0]
```

Comment:
Mmm, what about just `basic_lang = lang_codes[0]`? I think it would be more consistent... It could also be specified in the documentation (e.g. "the first language in the JSON list will be used as a fallback")